File: src/main/java/org/cloudio/morpheus/mail/RegisteredUserAdapter.java
package org.cloudio.morpheus.mail;

import java.util.Date;

/**
 * Created by zslajchrt on 24/08/15.
 */
public class RegisteredUserAdapter extends RegisteredUser implements MailOwner {
    public RegisteredUserAdapter() {
    }

    public RegisteredUserAdapter(RegisteredUser other) {
        super(other);
    }

    public String nick() {
        return getNick();
    }

    public String firstName() {
        return getFirstName();
    }

    public String lastName() {
        return getLastName();
    }

    public String email() {
        return getEmail();
    }

    public Date birthDate() {
        return getBirthDate();
    }
}

File: src/main/scala/org/cloudio/morpheus/mail/morpheus/RegisteredUserAdapter.scala
package org.cloudio.morpheus.mail.morpheus

import java.util.Date

import org.morpheus._
import org.morpheus.Morpheus._

/**
  * Created by zslajchrt on 24/08/15.
  */
@fragment
trait RegisteredUserAdapter extends MailOwner {
   this: RegisteredUser =>

  override def isMale: Boolean = regUserData.male

  override def nick: String = regUserData.nick

  override def lastName: String = regUserData.lastName

  override def email: String = regUserData.email

  override def birthDate: Date = regUserData.birthDate

  override def firstName: String = regUserData.firstName
}
